data/scenarios/Challenges/Ranching/fishing.yaml

```diff
@@ -0,0 +1,544 @@
+version: 1
+name: Fishing trip
+author: Karl Ostmo
+description: |
+  Catch a the appropriate seafood to make lunch.
+
+  If you haul any trash out of the lake, you
+  must throw it away in the enclosure to the southeast.
+  No littering!
+creative: false
+seed: 0
+attrs:
+  - name: shark
+    bg: "#2040bb"
+objectives:
+  - teaser: Hot rod
+    goal:
+      - |
+        `equip` your `fishing tackle`{=entity}, i.e. a rod and its accoutrements.
+    condition: |
+      as base {
+        equipped "fishing tackle";
+      }
+  - teaser: Sushi chain
+    id: lots_of_sushi
+    goal:
+      - |
+        Make 16 `california roll`{=entity}s
+    condition: |
+      as base {
+        rollCount <- count "california roll";
+        return $ rollCount >= 16;
+      }
+    prerequisite: made_sushi
+  - teaser: Sushi chef
+    id: made_sushi
+    goal:
+      - |
+        Make a `california roll`{=entity}
+    condition: |
+      as base {
+        has "california roll"
+      }
+  - teaser: Waste disposal
+    prerequisite:
+      logic:
+        and:
+          - lots_of_sushi
+          - not: littered
+    goal:
+      - |
+        Inventory must be free of all `junk`{=tag}.
+      - |
+        Dispose of `junk`{=tag} only in designated `rubbish enclosure`{=structure}s.
+      - |
+        Certain items can be burned to make room.
+        Good thing you brought your `torch`{=entity}!
+      - |
+        Otherwise, when the enclosure becomes full, the trash
+        hauler will come to empty it.
+    condition: |
+      // Returns true if prohibited item is in inventory.
+      def checkProhibited = \idx.
+        result <- tagmembers "junk" idx;
+        let totalCount = fst result in
+        let member = snd result in
+        let nextIdx = idx + 1 in
+
+        hasProhibited <- as base {has member};
+        if hasProhibited {
+          return false;
+        } {
+          if (nextIdx < totalCount) {
+            checkProhibited nextIdx;
+          } {
+            return true;
+          }
+        }
+        end;
+
+      checkProhibited 0;
+  - teaser: No littering
+    id: littered
+    hidden: true
+    optional: true
+    goal:
+      - |
+        You may dispose of `junk`{=tag} only in either of the designated `rubbish enclosure`{=structure}s.
```

Comment:
This is indeed less confusing now, but instead of being confusing it is simply cryptic.  None of the first few goals say anything about not littering, and if you happen to place litter on the ground it just immediately says "sorry, the scenario is not winnable!".  If you choose to keep playing after that and look at the goals, you can learn that you were not supposed to litter.  But players might not think to keep playing the scenario after losing.  I still think some of the text about not littering should also be copied into the "sushi chef" goal.

Reply:
Here's what it looks like after rebasing atop #1735.  Would that suffice?

![Screenshot from 2024-02-17 15-30-03](https://github.com/swarm-game/swarm/assets/261693/c2d85074-c786-4390-aefe-4187fdf23339)

Reply:
Yes, that looks great!  Except it should say "Catch the" not "Catch a the".